File: java/java-kafka-producer/src/main/java/com/showmeyourcode/kafka/java/kafka_producer/avro/ExampleUserRecord.java
package com.showmeyourcode.kafka.java.kafka_producer.avro;

public class ExampleUserRecord {
    private String name;
    private Long age;
    private String phoneNumber;

    public ExampleUserRecord(String name, Long age, String phoneNumber) {
        this.name = name;
        this.age = age;
        this.phoneNumber = phoneNumber;
    }

    public Long getAge() {
        return age;
    }

    public void setAge(Long age) {
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }
}

File: kafka-producer/src/main/kotlin/com/showmeyourcode/kafka/kafka_publisher/avro/ExampleUserRecord.kt
package com.showmeyourcode.kafka.kafka_publisher.avro

import com.github.avrokotlin.avro4k.AvroDoc
import kotlinx.serialization.Serializable

@Serializable
@AvroDoc("An example of Apache Avro schema")
data class ExampleUserRecord(val name: String, val age: Long, val phoneNumber: String? = null)
